pkg/manifest/manifest.go

```diff
@@ -0,0 +1,118 @@
+package manifest
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// Manifest stores Kubernetes object in Raw from a file.
+// It stores the GroupVersionKind for the manifest.
+type Manifest struct {
+	// OriginalFilename is set to the filename this manifest was loaded from.
+	// It is not guaranteed to be set or be unique, but we will set it when
+	// loading from disk to provide better debuggability.
+	OriginalFilename string
+
+	Raw []byte
+	GVK schema.GroupVersionKind
+
+	Obj *unstructured.Unstructured
+}
+
+// UnmarshalJSON unmarshals bytes of single kubernetes object to Manifest.
+func (m *Manifest) UnmarshalJSON(in []byte) error {
+	if m == nil {
+		return errors.New("Manifest: UnmarshalJSON on nil pointer")
+	}
+
+	// This happens when marshalling
+	// <yaml>
+	// ---	(this between two `---`)
+	// ---
+	// <yaml>
+	if bytes.Equal(in, []byte("null")) {
+		m.Raw = nil
+		return nil
+	}
+
+	m.Raw = append(m.Raw[0:0], in...)
+	udi, _, err := scheme.Codecs.UniversalDecoder().Decode(in, nil, &unstructured.Unstructured{})
+	if err != nil {
+		return errors.Wrapf(err, "unable to decode manifest")
+	}
+	ud, ok := udi.(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("expected manifest to decode into *unstructured.Unstructured, got %T", ud)
+	}
+
+	m.GVK = ud.GroupVersionKind()
+	m.Obj = ud.DeepCopy()
+	return nil
+}
+
+// Object returns underlying metav1.Object
+func (m *Manifest) Object() metav1.Object { return m.Obj }
```

Comment:
with this, why a public Obj?

Reply:
CVO accesses it directly in some cases:

https://github.com/jottofar/cluster-version-operator/blob/732ae625e991bf1447370b537caf9bcc22c850dc/pkg/cvo/cvo.go#L304

And in others Object() is used:

https://github.com/jottofar/cluster-version-operator/blob/732ae625e991bf1447370b537caf9bcc22c850dc/pkg/cvo/sync_worker.go#L918

Reply:
change CVO

Reply:
I removed the Object func